perfact/zodbsync/tests/test_sync.py

```diff
@@ -705,3 +705,305 @@ def test_extedit(self, encoding=None):
+            '/__root__/'+folder_1+'/'+s_folder_1+'/__meta__'
+        content = "[('title', '"+new_title+"'),('type', 'Folder'),]"
+        with open(path, 'w') as f:
+            f.write(content)
+
+        # playback changes and check if those are existent in zodb
+        self.run('playback', '/')
+        assert new_title == self.app.folder_1.s_folder_1.title
+
+        # wait for watch to notices played back changes
+        with open(path) as f:
+            meta = f.read()
+        self.watcher_step_until(watcher,
+                                lambda: "('title', '"+new_title+"')" in meta)
+
+    def test_commit_on_branch_and_exec_merge(self):
+        '''
+        change to a git feature branch and create a
+        structure there, commit it and change back to the master branch
+        on master branch check if changes from feature arent existent,
+        then merge feature branch and check if changes have been applied
+        correctly
+        '''
+
+        # change to feature branch and commit created folder/ subfolder
+        branch = "feature"
+        folder_1 = "folder_1"
+        s_folder_1 = "s_folder_1"
+        self.run('exec', 'git checkout -b {}'.format(branch))
+        self.app.manage_addFolder(id=folder_1)
+        self.app.folder_1.manage_addFolder(id=s_folder_1)
+        assert s_folder_1 in self.app.folder_1.objectIds()
+        self.run('record', '/')
+        assert os.path.isfile(
+            self.repo.path + '/__root__/'+folder_1+'/'+s_folder_1+'/__meta__'
+        )
+        self.gitrun('add', '-A')
+        self.gitrun('commit', '-m', 'test case 3')
+
+        # checkout to master and check that changes are not yet existent
+        self.run('exec', 'git checkout master')
+        assert os.path.isfile(
+            self.repo.path + '/__root__/'+folder_1+'/'+s_folder_1+'/__meta__'
+        ) is False
+        assert folder_1 not in self.app.objectIds()
+
+        # merge feature branch and check that changes are applied
+        self.run('exec', 'git merge {}'.format(branch))
+        assert os.path.isfile(
+            self.repo.path + '/__root__/'+folder_1+'/'+s_folder_1+'/__meta__'
+        )
+        assert folder_1 in self.app.objectIds()
+
+    def test_failing_playback_corrupt_metadata(self):
+        """
+        create a folder in zodb and record it,
+        write wrong meta data to the local file system, then playback
+        and check if an error occured
+        """
+
+        # create new folder and record it
+        folder_1 = "folder_1"
+        self.app.manage_addFolder(id=folder_1)
+        self.run('record', '/')
+
+        # break metadata
+        path = self.repo.path + '/__root__/'+folder_1+'/__meta__'
+        content = "[('gandalf', 'ThisIsAWrongKey'),]"
+        with open(path, 'w') as f:
+            f.write(content)
+
+        # test that playback fails
+        error = False
+        try:
+            self.run('playback', '/')
+        except KeyError:
```

Comment:
Please use `pytest.raises` for these, see [here](https://docs.pytest.org/en/6.2.x/assert.html)